### PR TITLE
feat(popups): auto-attach to all pages in Chromium

### PR DESCRIPTION
**Pull request by yury-s**
References: #1101

### CODE REVIEW

File: src/chromium/crPage.ts [hunk 1]
```diff
@@ -68,32 +68,35 @@ export class CRPage implements PageDelegate {
...
-      helper.addEventListener(this._client, 'Runtime.bindingCalled', event => this._onBindingCalled(event)),
-      helper.addEventListener(this._client, 'Runtime.consoleAPICalled', event => this._onConsoleAPI(event)),
-      helper.addEventListener(this._client, 'Runtime.exceptionThrown', exception => this._handleException(exception.exceptionDetails)),
-      helper.addEventListener(this._client, 'Runtime.executionContextCreated', event => this._onExecutionContextCreated(event.context)),
-      helper.addEventListener(this._client, 'Runtime.executionContextDestroyed', event => this._onExecutionContextDestroyed(event.executionContextId)),
-      helper.addEventListener(this._client, 'Runtime.executionContextsCleared', event => this._onExecutionContextsCleared()),
-      helper.addEventListener(this._client, 'Target.attachedToTarget', event => this._onAttachedToTarget(event)),
-      helper.addEventListener(this._client, 'Target.detachedFromTarget', event => this._onDetachedFromTarget(event)),
-    ];
+    // new Promise(r => setTimeout(r, 1000)).then(() => {
```
Comment (dgozman) [inline src/chromium/crPage.ts]: Commented code.

File: test/popup.spec.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ module.exports.describe = function({testRunner, expect, playwright, CHROMIUM, WE
   const {beforeAll, beforeEach, afterAll, afterEach} = testRunner;
 
   describe('window.open', function() {
-    it.fail(CHROMIUM)('should inherit user agent from browser context', async function({browser, server}) {
+    it('should inherit user agent from browser context', async function({browser, server}) {
```
Comment (dgozman) [inline test/popup.spec.js]: Let's add a similar test for anchor click.

File: src/chromium/crBrowser.ts [hunk 3]
```diff
@@ -43,10 +43,26 @@ export class CRBrowser extends platform.EventEmitter implements Browser {
...
   private _tracingClient: CRSession | undefined;
 
-  static async connect(transport: ConnectionTransport, slowMo?: number): Promise<CRBrowser> {
+  static async connect(transport: ConnectionTransport, isPersistent: boolean, slowMo?: number): Promise<CRBrowser> {
     const connection = new CRConnection(SlowMoTransport.wrap(transport, slowMo));
     const browser = new CRBrowser(connection);
-    await connection.rootSession.send('Target.setDiscoverTargets', { discover: true });
+    const session = connection.rootSession;
+    const promises = [session.send('Target.setDiscoverTargets', { discover: true })];
+    if (isPersistent) {
```
Comment (dgozman) [inline src/chromium/crBrowser.ts]: Let's clean this up a bit as discussed offline.

**Review by dgozman — APPROVED** [review]

File: src/chromium/crBrowser.ts (hunk 3 above)
Comment (yury-s) [reply src/chromium/crBrowser.ts]: Done.

File: src/chromium/crPage.ts (hunk 1 above)
Comment (yury-s) [reply src/chromium/crPage.ts]: Removed.

File: test/popup.spec.js (hunk 2 above)
Comment (yury-s) [reply test/popup.spec.js]: Done.

**Pull request merged**